app/controllers/course/admin/leaderboard_settings_controller.rb

```diff
@@ -17,12 +16,11 @@ def update #:nodoc:
   private
 
   def leaderboard_settings_params #:nodoc:
-    params.require(:leaderboard_settings).
+    params.require(:settings_leaderboard_component).
       permit(:title, :display_user_count, :enable_group_leaderboard, :group_leaderboard_title)
   end
 
-  # Load our settings adapter to handle leaderboard settings
-  def load_settings
-    @settings ||= Course::LeaderboardSettings.new(current_course.settings(:leaderboard))
+  def component
+    current_component_host[:course_leaderboard_component]
```

Comment:
Once concern about this (not this PR): we probably should let the component framework raise an error if the key like `course_leaderboard_component` is not defined.  Now that it returns `nil` and we cannot differentiate whether the component is disabled or we input the key wrongly (typo or sth).